### DISCUSSION TITLE
File preview

### DISCUSSION THREAD

**Discussion by Froc88** [rejetto/hfs#277] [General]
Hi!
Is there any possibility to add a preview of pdf, xlsx, docx files?
So you can see them without downloading them
Thank you!

**Comment by rejetto** [comment]
most browsers support pdf, so you can just "open", and it will open in the browser.
xlsx and docx normally are not supported by the browser, but potentially a plugin could add such capability.
Both chrome and firefox have plugins and you may find one that suits your needs.
Of course this implies that every browser should install that plugin.
The alternative is that the plugin is "for HFS", instead of "for the browser". There's no such plugin at the moment, but [someone could make it](https://github.com/rejetto/hfs/blob/main/dev-plugins.md) (even you). I may even consider making such plugin myself in the future, but don't know when.

0.46 (now in beta) offers a new command called "[show](https://github.com/rejetto/hfs/discussions/270)", that's just a different way of looking at files than the "open". There's also already plugin that expand the "show" to any file supported by the browser, like PDF.

**Reply by Froc88** [reply]
Thank you! I had not seen that version 0.46

**Reply by rejetto** [reply]
the show feature was born because people wanted next/previous buttons when watching pictures.
i recently found a little problem with it, will fix in next release.

**Reply by mitobin** [reply]
I downloaded .docx and .xlsx reader plugin on chrome browser but it doesn't work in user mode. is there any way to fix it?

**Reply by rejetto** [reply]
i don't know what you mean by user mode

**Reply by mitobin** [reply]
User mode is secure file sharing, must be logged in to access files. While sharing files in public mode with no account set, the plugin that reads .xlsx and .docx on chrome browser works properly.

**Reply by mitobin** [reply]
Is there any way to fix the above problem? Please consider this issue

**Reply by rejetto** [reply]
with user mode you mean that in HFS you restrict permissions to a file?

**Reply by mitobin** [reply]
Yes, in HFS. When sharing files in user mode, even though you have set up admin rights and do not limit file permissions, the chrome browser plugin still cannot read it. When I set the file sharing mode to public, without using an account, the plugin still works normally

**Reply by rejetto** [reply]
link me the chrome extension you are using

**Reply by mitobin** [reply]
I use this plugin: https://chrome.google.com/webstore/detail/docs-online-viewer/gmpljdlgcdkljlppaekciacdmdlhfeon?hl=vi

**Reply by rejetto** [reply]
well, that plugin/method gives your file to google.
If a file is password protected then you probably don't want to give it to google (or any other company).
I cannot consider integrating this solution.

**Reply by mitobin** [reply]
Unfortunately there is no solution to this problem. I use .pdf reader plugin for chrome (https://chrome.google.com/webstore/detail/pdf-viewer/oemmndcbldboiebfnladdacbdfmadadm?hl=vi), it works properly with user mode. But the plugin reads .docx and xlsx not working even though logged in and gave full access to the files

**Reply by rejetto** [reply]
chrome supports pdf without any plugin

**Reply by rejetto** [reply]
these plugins don't work because wants your file to be uploaded (automatically) to a server.
A better plugin would do the job locally, no server involved. I don't know if such plugin exists, but it's technically possible.
The same technical possibility is available in HFS.

**Reply by mitobin** [reply]
Preview without downloading document formats is very convenient for work. hopefully this feature will be available on HFS soon. Thank you very much
